File: 2580-circular-sentence/circular-sentence.cpp
class Solution {
public:
    bool isCircularSentence(string s) {
        int sz = s.length();

        if(s[0] != s[sz-1]) return false;

        for(int i=0; i<sz; i++)
            if(s[i] == ' ' && s[i-1] != s[i+1]) return false;

        return true;
    }
};

File: 2580-circular-sentence/circular-sentence.java
class Solution {
    public boolean isCircularSentence(String sentence) {
        int sz = sentence.length();
        char last = '/';

        for(int i=0; i<sz; i++){
            if(i-1 >= 0 && sentence.charAt(i-1) == ' '){
                if(sentence.charAt(i) != last) return false;
            }
            if(sentence.charAt(i) != ' '){
                last = sentence.charAt(i);
            }
        }

        return last == sentence.charAt(0);
    }
}